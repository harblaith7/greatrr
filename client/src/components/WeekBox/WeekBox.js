import React, { Component } from 'react';
import "./WeekBox.scss"



class WeekBox extends Component {
    render() {
        const {weekAbbreviation, id, weekStatus} = this.props
        return (
            <>
                {weekStatus && (
                    <div className={`WeekBox ${weekStatus[id] && "WeekBox--active"}`}>
                        {weekAbbreviation}
                    </div>
                )}
            </>
        );
    }
}

export default WeekBox;