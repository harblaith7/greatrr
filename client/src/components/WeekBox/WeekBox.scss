@import "../../sass/variables.scss";

.WeekBox {
    width: 5rem;
    height: 5rem;
    background-color: $inactive-blue;
    border-radius: 0.3rem;
    display: flex;
    align-items: center;
    justify-content: center;
    cursor: pointer;
    box-shadow: 0.1rem 0.1rem 1rem $very-light-black;
    font-size: 1.75rem;
    font-weight: 700;
    color: rgba(56, 55, 55, 0.795);

    &--active {
        background-color: $active-green;
    }
}