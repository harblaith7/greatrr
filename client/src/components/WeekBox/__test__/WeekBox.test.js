import React from "react"
import {render, cleanup, fireEvent} from "@testing-library/react"
 
import WeekBox from "../WeekBox"



afterEach(cleanup)


test("Tests if div contains correct week abbreviation", () => {  

    const wrapper = render(
        <WeekBox
            weekAbbreviation = "M"
            weekId = {0}
            weekStatus = {[false, false, false, false, false, false, false]}
            testId={'M-0'}
        />
    )

    let weekBox = wrapper.getByTestId("M-0")

    setTimeout(() => {
        expect(weekBox.textContent).toBe("M")
    }, 2000)
})

test("Renders", () => {
    const {asFragment} = render(
        <WeekBox
            weekAbbreviation = "M"
            weekId = {0}
            weekStatus = {[false, false, false, false, false, false, false]}
            testId={'M-0'}
        />
    )

    expect(asFragment()).toMatchSnapshot()
})

/*
test("Added active class", () => {

    const {getByTestId} = render(
        <WeekBox
            weekAbbreviation = "M"
            weekId = {0}
            weekStatus = {[false, false, false, false, false, false, false]}
        />
    )

    fireEvent.click(getByTestId("week-box"))
})
*/
