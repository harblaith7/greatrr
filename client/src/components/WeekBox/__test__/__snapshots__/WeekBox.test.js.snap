// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders 1`] = `
<DocumentFragment>
  <div
    class="WeekBox "
    data-testid="M-0"
  >
    M
  </div>
</DocumentFragment>
`;
